sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -733,7 +733,8 @@ def test_kneighbors_regressor_sparse(n_samples=40,
         knn.fit(sparsemat(X), y)
         for sparsev in SPARSE_OR_DENSE:
             X2 = sparsev(X)
-            assert_true(np.mean(knn.predict(X2).round() == y) > 0.95)
+            assert_raises(ValueError, knn.predict, csr_matrix(X2))
```

Comment:
This should be a red flag: changing a previously working behavior to an ``assert_raises`` means we broke someones code ;)

Reply:
Got it, I might have gotten a little overzealous. I'll edit this to only check for the error for sparse matrices with precomputed kernels (which we're deciding not to support)